### PR TITLE
DOCSP-39091  Recommend that users upgrade to the latest server patch

### PR DESCRIPTION
**Pull request by ltran-mdb2**
## DESCRIPTION

Recommend that users upgrade to the latest server patch prior to migration

## STAGING

https://preview-mongodbltranmdb2.gatsbyjs.io/cluster-sync/DOCSP-39091/release-notes/1.7/#minimum-supported-version
https://preview-mongodbltranmdb2.gatsbyjs.io/cluster-sync/DOCSP-39091/connecting/onprem-to-onprem/
https://preview-mongodbltranmdb2.gatsbyjs.io/cluster-sync/DOCSP-39091/connecting/onprem-to-atlas/

## JIRA

https://jira.mongodb.org/browse/DOCSP-39091

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=66312de3b67dbf804aa92668

## SELF-REVIEW CHECKLIST

- [ ] Does each file have 3-5 taxonomy facet tags?
  See the [taxonomy tagging instructions](https://wiki.corp.mongodb.com/display/DE/Taxonomy+tagging+instructions) and this [example PR](https://github.com/10gen/cloud-docs/pull/5042/files) 
- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## EXTERNAL REVIEW REQUIREMENTS

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/includes/migration-upgrade-recommendation.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+For best performance, upgrade your source and destination clusters to
+the most recent server patch release prior to migration. For more
+information, see :ref:`MongoDB Patch Releases <versions-patch-release>`.
```
Comment (jeff-allen-mongo) [inline source/includes/migration-upgrade-recommendation.rst:3]: Maybe this page might be slightly better to link to:

https://www.mongodb.com/docs/manual/tutorial/upgrade-revision/

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM % a suggestion!

File: source/includes/migration-upgrade-recommendation.rst (hunk 1 above)
Comment (ltran-mdb2) [reply source/includes/migration-upgrade-recommendation.rst]: Done! Thanks for this suggestion.

File: source/connecting/onprem-to-atlas.txt [hunk 2]
```diff
@@ -21,6 +21,10 @@ Atlas cluster using {+c2c-full-product-name+}.
 For additional details on connecting to an Atlas cluster, see
 :ref:`Connect to a Database Deployment <atlas-connect-to-deployment>`
 
+.. note::
+
+   .. include:: /includes/migration-upgrade-recommendation.rst
```
Comment (rohshar) [inline source/connecting/onprem-to-atlas.txt:26]: For on prem to atlas, the atlas cluster will be updated by mongodb in a rolling manner, so they only need to worry about the on-prem cluster. I think we should make that clear.

**Review by rohshar — CHANGES_REQUESTED** [review]

File: source/connecting/onprem-to-atlas.txt (hunk 2 above)
Comment (ltran-mdb2) [reply source/connecting/onprem-to-atlas.txt]: Done!

**Comment by ltran-mdb2** [comment]
Thanks for your feedback @rohshar ! Can you take another look when you get a chance?

File: source/connecting/onprem-to-atlas.txt (hunk 2 above)
Comment (rohshar) [reply source/connecting/onprem-to-atlas.txt]: The atlas cluster can either be the source or destination. I think that we should specify that the on prem cluster is the one that needs to be updated.

Comment (ltran-mdb2) [reply source/connecting/onprem-to-atlas.txt]: Fixed!

File: source/connecting/onprem-to-atlas.txt [hunk 3]
```diff
@@ -21,6 +21,12 @@ Atlas cluster using {+c2c-full-product-name+}.
 For additional details on connecting to an Atlas cluster, see
 :ref:`Connect to a Database Deployment <atlas-connect-to-deployment>`
 
+.. note::
+
+   For best performance, upgrade your on self-managed cluster to the
+   most recent server patch release prior to migration. For more
```
Comment (mdb-ashley) [inline source/connecting/onprem-to-atlas.txt:27]:
```suggestion
   most recent MongoDB Server patch release prior to migration. For more
```

File: source/includes/migration-upgrade-recommendation.rst [hunk 4]
```diff
@@ -0,0 +1,3 @@
+For best performance, upgrade your source and destination clusters to
+the most recent server patch release prior to migration. For more
```
Comment (mdb-ashley) [inline source/includes/migration-upgrade-recommendation.rst:2]:
```suggestion
the most recent MongoDB Server patch release prior to migration. For more
```

File: source/connecting/onprem-to-atlas.txt (hunk 3 above)
Comment (ltran-mdb2) [reply source/connecting/onprem-to-atlas.txt]: Done!

File: source/includes/migration-upgrade-recommendation.rst (hunk 4 above)
Comment (ltran-mdb2) [reply source/includes/migration-upgrade-recommendation.rst]: Done!

**Pull request merged**